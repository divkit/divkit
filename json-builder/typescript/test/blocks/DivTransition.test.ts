import {
    DivAppearanceSetTransition,
    DivChangeBoundsTransition,
    DivFadeTransition,
    DivImage,
    DivScaleTransition,
    DivState,
    DivText,
    fixed,
} from '../../src';

describe('DivTransitions tests', (): void => {
    it('should create block with in/out transitions', (): void => {
        const block = new DivText({
            id: 'id',
            text: 'Text with transitions',
            transition_triggers: ['visibility_change'],
            transition_in: new DivAppearanceSetTransition({
                items: [
                    new DivFadeTransition({
                        alpha: 0,
                        duration: 250,
                        start_delay: 250,
                    }),
                    new DivScaleTransition({
                        scale: 2,
                        duration: 500,
                        interpolator: 'ease_in_out',
                    }),
                ],
            }),
            transition_out: new DivScaleTransition({
                scale: 0,
                duration: 500,
                interpolator: 'linear',
            }),
        });

        expect(block).toMatchSnapshot();
    });

    it('should create block with change transitions', (): void => {
        const block = new DivState({
            div_id: 'id',
            states: [
                {
                    state_id: 'state1',
                    div: new DivImage({
                        id: 'state1/image',
                        image_url: 'https://image.url',
                        width: fixed(100),
                        height: fixed(50),
                        transition_triggers: ['state_change'],
                        transition_change: new DivChangeBoundsTransition({
                            duration: 500,
                            interpolator: 'ease_in',
                        }),
                    }),
                },
                {
                    state_id: 'state2',
                    div: new DivImage({
                        id: 'state2/image',
                        image_url: 'https://image.url',
                        width: fixed(50),
                        height: fixed(100),
                        transition_triggers: ['state_change'],
                        transition_change: new DivChangeBoundsTransition({
                            duration: 500,
                            interpolator: 'ease_out',
                        }),
                    }),
                },
            ],
        });

        expect(block).toMatchSnapshot();
    });
});
