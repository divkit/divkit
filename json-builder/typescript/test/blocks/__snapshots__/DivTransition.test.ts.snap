// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DivTransitions tests should create block with change transitions 1`] = `
{
  "type": "state",
  "div_id": "id",
  "states": [
    {
      "state_id": "state1",
      "div": {
        "type": "image",
        "height": {
          "type": "fixed",
          "value": 50
        },
        "id": "state1/image",
        "image_url": "https://image.url",
        "transition_change": {
          "type": "change_bounds",
          "duration": 500,
          "interpolator": "ease_in"
        },
        "transition_triggers": [
          "state_change"
        ],
        "width": {
          "type": "fixed",
          "value": 100
        }
      }
    },
    {
      "state_id": "state2",
      "div": {
        "type": "image",
        "height": {
          "type": "fixed",
          "value": 100
        },
        "id": "state2/image",
        "image_url": "https://image.url",
        "transition_change": {
          "type": "change_bounds",
          "duration": 500,
          "interpolator": "ease_out"
        },
        "transition_triggers": [
          "state_change"
        ],
        "width": {
          "type": "fixed",
          "value": 50
        }
      }
    }
  ]
}
`;

exports[`DivTransitions tests should create block with in/out transitions 1`] = `
{
  "type": "text",
  "id": "id",
  "text": "Text with transitions",
  "transition_in": {
    "type": "set",
    "items": [
      {
        "type": "fade",
        "alpha": 0,
        "duration": 250,
        "start_delay": 250
      },
      {
        "type": "scale",
        "duration": 500,
        "interpolator": "ease_in_out",
        "scale": 2
      }
    ]
  },
  "transition_out": {
    "type": "scale",
    "duration": 500,
    "interpolator": "linear",
    "scale": 0
  },
  "transition_triggers": [
    "visibility_change"
  ]
}
`;
